kratos/utilities/sparse_matrix_multiplication_utility.h

```diff
@@ -326,6 +320,7 @@ class SparseMatrixMultiplicationUtility
         IndexType* c_ptr = new IndexType[nrows + 1];
         c_ptr[0] = 0;
 
+        // TODO: Replace with block_for_each
```

Comment:
a while ago, I mentioned prolonging the lifetime of TLS so that the main thread can access the results and do some fancy complex reduction on them. That would be perfect here.